internal/bgp/frr/frr_test.go

```diff
@@ -1086,3 +1133,174 @@ func TestLoggingConfigurationOverrideByEnvironmentVar(t *testing.T) {
 	sessionManager.reloadConfig <- reloadEvent{config: config}
 	testCheckConfigFile(t)
 }
+
+func TestAddToAdvertisements(t *testing.T) {
+	tests := []struct {
+		name      string
+		current   []*advertisementConfig
+		toAdd     *advertisementConfig
+		expected  []*advertisementConfig
+		shouldErr bool
+	}{
+		{
+			name:    "starting empty",
+			current: []*advertisementConfig{},
+			toAdd: &advertisementConfig{
+				Prefix:   "192.168.1.1/32",
+				IPFamily: ipfamily.IPv4,
+			},
+			expected: []*advertisementConfig{
+				{
+					Prefix:   "192.168.1.1/32",
+					IPFamily: ipfamily.IPv4,
+				},
+			},
+		},
+		{
+			name: "mismatch localpref",
```

Comment:
wouldn't we want to test "mismatch IPFamily" as well?

Reply:
I agree, but this is a backport, so we don't normally change the content as it was already reviewed and approved upstream and in the various openshift versions.

Reply:
lgtm other than that, didn't notice it's a backport